pkg/build/ignore/docker/dockign.go

```diff
@@ -0,0 +1,116 @@
+package docker
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DOCKIGN struct {
+	Config *api.Config
+}
+
+func New(req *api.Config) *DOCKIGN {
+	d := &DOCKIGN{
+		Config: req,
+	}
+	return d
+}
+
+func (b *DOCKIGN) Ignore(config *api.Config) {
```

Comment:
there's no reason for this to be a method on a struct if you're going to take the only field of the struct as an arg anyway....  either make it a first class function and drop the struct, or have it take no args.

What you really probably want to do here is define an interface with a single Ignore() method so you can provide a mock implementation for testing.  And again you can either take an arg on that method, or you can simply have the implementation be defined on a struct that includes the Config object....but the former makes more sense to me.  You're really just defining an interface for a utility function and the implementation of the utility function doesn't need to hold any state.

on top of that i'd split this out into two methods:
1) get the list of files to ignore
2) remove files in a list from a directory